Remotion/Validation/UnitTests/Validators/NotEmptyCollectionValidatorTest.cs

```diff
@@ -0,0 +1,261 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using Remotion.Validation.Implementation;
+using Remotion.Validation.Validators;
+
+namespace Remotion.Validation.UnitTests.Validators
+{
+  [TestFixture]
+  public class NotEmptyCollectionValidatorTest : ValidatorTestBase
+  {
+    private class Enumerable<T> : IEnumerable<T>
+    {
+      private readonly List<T> _list = new();
+
+      public IEnumerator<T> GetEnumerator ()
+      {
+        return _list.GetEnumerator();
+      }
+
+      IEnumerator IEnumerable.GetEnumerator ()
+      {
+        return GetEnumerator();
+      }
+    }
+
+    private class SimpleCollection : ICollection
+    {
+      private readonly List<object> _list = new();
+
+      public IEnumerator GetEnumerator ()
+      {
+        return _list.GetEnumerator();
+      }
+
+      public void CopyTo (Array array, int index)
+      {
+        var objectArray = array.Cast<object>().ToArray();
+        _list.CopyTo(objectArray, index);
+      }
+
+      public void Add (object item)
+      {
+        _list.Add(item);
+      }
+
+      public int Count => _list.Count;
+      public bool IsSynchronized => true;
+      public object SyncRoot { get; } = new();
+    }
+
+    private class GenericCollection<T> : ICollection<T>
+    {
+      private readonly List<T> _list = new();
+
+      public IEnumerator GetEnumerator ()
+      {
+        return _list.GetEnumerator();
+      }
+
+      public int Count => _list.Count;
+
+      IEnumerator<T> IEnumerable<T>.GetEnumerator ()
+      {
+        return _list.GetEnumerator();
+      }
+
+      public void Add (T item)
+      {
+        _list.Add(item);
+      }
```

Comment:
Syntax-question: do you want to shorten those oneliners using expression-bodied member syntax? (Resharper has a quick fix for that)

It's also okay to put them at the end of the file. That's become something we do in the tests.

```suggestion
      public void Add (T item) => _list.Add(item);
```